manifests/plugin/versionlock.pp

```diff
@@ -0,0 +1,30 @@
+# Class: zypprepo::plugin::versionlock
+#
+# @summary This class sets the structure for the lock file
+#
+# @param ensure
+#   specifies if versionlock should be present or absent. Defaults to present.
+#
+# @param path
+#   Absolute path to the Zypper locks file. Defaults /etc/zypp/locks.
+#
+# @example Sample usage:
+#   include zypprepo::plugin::versionlock
+#
+class zypprepo::plugin::versionlock (
+  Enum['present', 'absent'] $ensure = 'present',
```

Comment:
Why this parameter without using it ?
IMO this parameter should be simply removed.
Otherwise it may suggest that it's possible to make sure it's not there.

Reply:
Because of how concat is used ensure=>absent will indeed remove the lock. It can be argued that we can remove the statement as this is the same as ensure=>absent. However, will that also not break user's expectations (i.e. to ensure this is absent add ensure=>absent)?

Reply:
Removing a statement does not imply the absence. It just means that we do not describe the desired state of this. So this means that it is not handled by Puppet.
Arguing that remove a statement is same that `ensure => absent` is misunderstanding how Puppet works.

Reply:
So I double checked my test just to be certain. 

Due to how concat is setup, it will remove any line that is not specifically defined by the module. This is unlike augeas that will remove specific lines. I can, of course, remove the ensure parameter, but would this break user's expectations?

Reply:
@dhollinger and @Dan33l , how about this as a plan?
1. I remove the ensure block as only present does anything. And
2. I add a notice to the documentation giving the warning that once any locks are defined by the module, all locks must be defined by the module or they will be removed.

Does this work?